core/domain/collection_jobs_one_off_test.py

```diff
@@ -0,0 +1,153 @@
+
+from core.domain import collection_domain
+from core.domain import collection_jobs_one_off
+from core.domain import collection_services
+from core.domain import rights_manager
+from core.platform import models
+from core.tests import test_utils
+import feconf
+
+(job_models, collection_models,) = models.Registry.import_models([
+    models.NAMES.job, models.NAMES.collection])
+search_services = models.Registry.import_search_services()
+
+
+class CollectionMigrationJobManagerTest(test_utils.GenericTestBase):
+
+    ALBERT_EMAIL = 'albert@example.com'
+    ALBERT_NAME = 'albert'
+
+    COLLECTION_ID = 'collection_id'
+    EXP_ID = 'exp_id'
+
+    def setUp(self):
+        super(CollectionMigrationJobManagerTest, self).setUp()
+
+        # Setup user who will own the test collections.
+        self.albert_id = self.get_user_id_from_email(self.ALBERT_EMAIL)
+        self.signup(self.ALBERT_EMAIL, self.ALBERT_NAME)
+        self.process_and_flush_pending_tasks()
+
+    def test_migration_job_does_not_convert_up_to_date_collection(self):
+        """Tests that the collection migration job does not convert an
+        collection that is already the latest collection content schema version.
+        """
+        # Create a new, default exploration that should not be affected by the
+        # job.
+        collection = collection_domain.Collection.create_default_collection(
+            self.COLLECTION_ID, 'A title', 'A Category', 'An Objective')
+        collection_services.save_new_collection(self.albert_id, collection)
+        self.assertEqual(
+            collection.schema_version,
+            feconf.CURRENT_COLLECTION_SCHEMA_VERSION)
+        yaml_before_migration = collection.to_yaml()
+
+        # Start migration job.
+        job_id = (
+            collection_jobs_one_off.CollectionMigrationJobManager.create_new())
+        collection_jobs_one_off.CollectionMigrationJobManager.enqueue(job_id)
+        self.process_and_flush_pending_tasks()
+
+        # Verify the collection is exactly the same after migration.
+        updated_collection = (
+            collection_services.get_collection_by_id(self.COLLECTION_ID))
+        self.assertEqual(
+            updated_collection.schema_version,
+            feconf.CURRENT_COLLECTION_SCHEMA_VERSION)
+        after_converted_yaml = updated_collection.to_yaml()
+        self.assertEqual(after_converted_yaml, yaml_before_migration)
+
+    def test_migration_job_skips_deleted_collection(self):
+        """Tests that the collection migration job skips deleted collection
+        and does not attempt to migrate.
+        """
+        collection = collection_domain.Collection.create_default_collection(
+            self.COLLECTION_ID, 'A title', 'A Category', 'An Objective')
+        collection_services.save_new_collection(self.albert_id, collection)
+
+        # Note: This creates a summary based on the upgraded model (which is
+        # fine). A summary is needed to delete the exploration.
```

Comment:
I think the deletion function has commands for both deleting the summary and deleting the exploration itself. So, in order to avoid special-casing, this test creates a summary object too (otherwise the deletion function will throw an exception).